src/components/shared/WithSize.js

```diff
@@ -37,17 +37,34 @@ export function withSize<
 > {
   return class WithSizeWrapper extends React.PureComponent<*, State> {
     _resizeListener: () => void;
+    _visibilitychangeListener: () => void;
+    _isSizeInfoDirty: boolean = false;
     state = { width: 0, height: 0 };
 
     componentDidMount() {
       const container = findDOMNode(this); // eslint-disable-line react/no-find-dom-node
       if (!container) {
         throw new Error('Unable to find the DOMNode');
       }
+
       this._resizeListener = () => {
-        this._updateWidth(container);
+        if (!document.hidden) {
+          this._updateWidth(container);
+        } else {
+          this._isSizeInfoDirty = true;
+        }
+      };
+      this._visibilitychangeListener = () => {
```

Comment:
Can you add some documentation to `_resizeListener` and `_visibilitychangeListener` that explains how the size only updates when the document is visible? This will help future contributors understand what is going on in this code.

Reply:
I think this method has gotten complicated enough with these changes that it should be broken up a bit. Initializing functions in a method like this is a bit risky, because if the ordering of method calls got messed up, the functions wouldn't exist, and if you called them, they would throw an error. I think it would be safer and more maintainable if we broke them out into [class property arrow functions](https://babeljs.io/docs/en/next/babel-plugin-proposal-class-properties.html). These can be safely passed to an event handler without running `.bind(this)` on them.

e.g.

Here is an example diff of what I'm talking about.

```diff
diff --git a/src/components/shared/WithSize.js b/src/components/shared/WithSize.js
index 46cc5a77..d8f08030 100644
--- a/src/components/shared/WithSize.js
+++ b/src/components/shared/WithSize.js
@@ -36,30 +36,16 @@ export function withSize<
   $ReadOnly<$Diff<Props, SizeProps>>
 > {
   return class WithSizeWrapper extends React.PureComponent<*, State> {
-    _resizeListener: () => void;
-    _visibilitychangeListener: () => void;
     _isSizeInfoDirty: boolean = false;
     state = { width: 0, height: 0 };
+    _container: ?(Element | Text);
 
     componentDidMount() {
       const container = findDOMNode(this); // eslint-disable-line react/no-find-dom-node
       if (!container) {
         throw new Error('Unable to find the DOMNode');
       }
-
-      this._resizeListener = () => {
-        if (!document.hidden) {
-          this._updateWidth(container);
-        } else {
-          this._isSizeInfoDirty = true;
-        }
-      };
-      this._visibilitychangeListener = () => {
-        if (!document.hidden && this._isSizeInfoDirty) {
-          this._updateWidth(container);
-          this._isSizeInfoDirty = false;
-        }
-      };
+      this._container = container;
       window.addEventListener('resize', this._resizeListener);
       window.addEventListener(
         'visibilitychange',
@@ -71,6 +57,29 @@ export function withSize<
       requestAnimationFrame(() => this._updateWidth(container));
     }
 
+    _resizeListener = () => {
+      const container = this._container;
+      if (!container) {
+        return;
+      }
+      if (!document.hidden) {
+        this._updateWidth(container);
+      } else {
+        this._isSizeInfoDirty = true;
+      }
+    };
+
+    _visibilitychangeListener = () => {
+      const container = this._container;
+      if (!container) {
+        return;
+      }
+      if (!document.hidden && this._isSizeInfoDirty) {
+        this._updateWidth(container);
+        this._isSizeInfoDirty = false;
+      }
+    };
+
     componentWillUnmount() {
       window.removeEventListener('resize', this._resizeListener);
       window.removeEventListener(
```

Notice how if someone erroneously called the `_visibilityChangeListener()` it won't throw an error, as it checks for the existence of the `container`.